alakum23/CEN3031GroupProj
# Initial HazardVis Project Setup

# Issue
[Related JIRA Issue One](https://cen3031team22.atlassian.net/browse/CT2-27?atlOrigin=eyJpIjoiOTg2MTEwYmNlYzM3NDQwZThmZjg3ZDAyZDc0MDJjNDMiLCJwIjoiaiJ9)

[Related JIRA Issue Two](https://cen3031team22.atlassian.net/browse/CT2-28?atlOrigin=eyJpIjoiZDFjOGM1OGYxMWM3NGEwMTk2MTNlNmFlZDZmMmQ3ZjMiLCJwIjoiaiJ9)

[Related JIRA Issue Three](https://cen3031team22.atlassian.net/browse/CT2-29?atlOrigin=eyJpIjoiNzM5MDQ5N2I3NjE5NGU1NTk4ODNiMDhjYmRiNTZhMGQiLCJwIjoiaiJ9)

[Related JIRA Issue Four](https://cen3031team22.atlassian.net/browse/CT2-9?atlOrigin=eyJpIjoiYjlkM2VlZjIyYjhhNDExNGEyYTI2NmJlNzg2Mjc3Y2EiLCJwIjoiaiJ9)

# Description
Created initial project folder structure
Created sample cesium viewer page
Created initial backend server
Created JavaScript ESLinting
Created JavaScript unit testing using JEST
Created Building and Testing NPM scripts
Updated CircleCI pipeline to use NODE JS and webpack builds
Updated SonarCloud to utilize unit test coverage reports
Created README documentation

# Type Of Change

# Test Steps
I ran each new npm script and found they emitted no errors (besides those already expected). The backend server was tested and works as expected in both development and production mode. The client website works as expected in the browser. 

I tested the setup process on two different machines with success for both systems.

One currently known bug is that using the angled 2D map view cesium provides emits an error. 
Another bug is that the results of npm run lint are currently not displaying on circleci.
There are no other known bugs.

# Checklist
